src/main/java/org/snomed/snowstorm/core/data/services/RefsetDescriptorUpdaterService.java

```diff
@@ -84,19 +83,19 @@ public void preCommitCompletion(Commit commit) throws IllegalStateException {
 		String branchPath = commit.getBranch().getPath();
 		for (SearchHit<QueryConcept> searchHit : searchHits.getSearchHits()) {
 			long conceptIdL = searchHit.getContent().getConceptIdL();
-			if (Concepts.REFSET.equals(String.valueOf(conceptIdL))) {
-				// Edge case where first importing.
-				continue;
-			}
+			String conceptIdS = String.valueOf(conceptIdL);
 
-			doAddToRefset(commit, conceptIdL, branchPath);
+			List<ReferenceSetMember> members = referenceSetMemberService.findMembers(branchPath, new MemberSearchRequest().referencedComponentId(conceptIdS).referenceSet(Concepts.REFSET_DESCRIPTOR_REFSET), PAGE_REQUEST).getContent();
+			if (members.isEmpty()) {
+				doAddMemberToRefset(commit, conceptIdL, branchPath);
+			}
```

Comment:
There's a number of changes within the PR, but essentially it all boils down to this block here. If there are no members for the concept in the reference set, then some will be added. This will prevent duplicates.